pa4/checker-pa4.ml

```diff
@@ -122,29 +123,39 @@ and case_element =
 exception MethodRedefineError of string;;
 exception AttributeRedefineError of string;;
 
-let dummy_exp_body = {line_number= "0"; expression_type= String(""); static_type= None;}
-let int_exp_body = {line_number= "0"; expression_type= Integer("0"); static_type= None;}
-let string_exp_body = {line_number= "0"; expression_type= String(""); static_type= None;}
+let int_static_type = (Option.some (Class "Int"));;
+let self_type_static_type = (Option.some (Class "SELF_TYPE"));;
+let str_static_type = (Option.some (Class "String"));;
+let obj_static_type = (Option.some (Class "Object"));;
+let obj_abort_exp_body = {line_number= "0"; expression_type=Internal(("0", "Object"), "Object", "abort") ; static_type=obj_static_type;}
```

Comment:
Could you include a comment above this stating that this is the expression body for built-in methods? It wasn't immediately clear to me what this was for at first, so that would help.